### PR TITLE
Fix: RAC4649: Jenkins is not updating PR quality gate status in PR

### PR DESCRIPTION
**Pull request by PengTian0**
Fix  [RAC4649](https://rackhd.atlassian.net/browse/RAC-4649)
There are 2 causes of this issue:
1. For pull request which has merge conflict, there is no manifest file generated.
    So the stage write back won't add comments to pull request or set the status of it.
2. If there is no test report generated, the post-result.py will add comments to pull request and exit with 1.      
    That leads to the shell script write_back_github.sh exit with 1 and won't execute the script commit_status_setter.py

The pull requests affected by this issue:
https://github.com/RackHD/on-build-config/pull/131

### CODE REVIEW

**Comment by PengTian0** [comment]
test this please

**Comment by PengTian0** [comment]
@anhou @yyscamper @panpan0000 @changev 
@amymullins @keedya

File: build-release-tools/application/pr_parser.py [hunk 1]
```diff
@@ -103,14 +105,7 @@ def parse_pr(self, base_repo, base_pull_id):
...
                     sys.exit(1)
-                try:
-                    dep_pr = gh.get_repo(repo).get_pull(long(pull_id))
-                    if not dep_pr.mergeable:
-                        print "ERROR: the pr of {0} is unmergeable.\n{1}".format(dep_pr_url, pr.mergeable_state)
-                        sys.exit(1)
-                    sha = 'origin/pr/{0}/merge'.format(pull_id)
-                except Exception as error:
-                    print "ERROR: the pr of {0} doesn't exist.\n{1}".format(dep_pr_url, error)
+                sha = 'origin/pr/{0}/merge'.format(pull_id)
```
Comment (anhou) [inline build-release-tools/application/pr_parser.py]: what does 'pull_id' mean? pull request id? naming  could be more meanful

File: build-release-tools/application/pr_parser.py [hunk 2]
```diff
@@ -210,36 +207,54 @@ def get_latest_commit(self, repo, branch):
...
+        is_mergable = True
+        all_prs = self.get_all_related_prs(self.__repo, self.__merge_commit_sha, self.__pull_id)
+        for pr in all_prs:
+            repo, sha, pull_id = pr
+            pr = gh.get_repo(repo).get_pull(long(pull_id))
+            if not pr.mergeable:
+                is_mergable = False
+                print("ERROR: the pr {0} of {1} is unmergeable.\n{1}".format(pull_id, repo,  pr.mergeable_state))
+
+        return is_mergable
```
Comment (anhou) [inline build-release-tools/application/pr_parser.py]: if pr not in all_prs,   True would be returned is expected?

File: build-release-tools/application/pr_parser.py (hunk 1 above)
Comment (PengTian0) [reply build-release-tools/application/pr_parser.py]: Thanks!
Changed to ```pr_number```

**Review by anhou — APPROVED** [review]

**Comment by panpan0000** [comment]
👍

**Pull request merged**